src/main/java/com/users/service/UserService.java

```diff
@@ -70,4 +73,19 @@ private User findById(Long id) {
         return userRepository.findById(id).
                 orElseThrow(() -> new EntityNotFoundException(String.format("User with id %s doesn't exist", id)));
     }
+
+    public UserDto findByEmail(String email) {
+        User byEmail = userRepository.findByEmail(email);
+        return userMapper.toDto(byEmail);
+    }
+
+    public List<UserDto> findByDateOfBirth(Instant dateOfBirth) {
+        List<User> byDateOfBirth = userRepository.findByDateOfBirth(dateOfBirth);
+        return userMapper.toDto(byDateOfBirth);
+    }
+
+    public List<UserDto> findAllSpecification(Specification<User> userSpecification) {
```

Comment:
specificatia se creaza aici nu in controller

Reply:
inteles